### PR TITLE
replace mysql driver, modify error handling

### PR DESCRIPTION
**Pull request by yseto**
replace mysql driver
modify error handling
use net.JoinHostPort

### CODE REVIEW

File: lib/mysql.go [hunk 1]
```diff
@@ -1213,7 +1250,7 @@ func Do() {
 	optInnoDB := flag.Bool("disable_innodb", false, "Disable InnoDB metrics")
 	optMetricKeyPrefix := flag.String("metric-key-prefix", "mysql", "metric key prefix")
 	optEnableExtended := flag.Bool("enable_extended", false, "Enable Extended metrics")
-	optDebug := flag.Bool("debug", false, "Print debugging logs to stderr")
+	flag.Bool("debug", false, "Print debugging logs to stderr(deprecated)") // backward compatibility
```
Comment (Arthur1) [inline lib/mysql.go:1253]: It says "deprecated," but has the debug mode already been obsoleted?

File: lib/mysql.go [hunk 2]
```diff
@@ -384,22 +412,25 @@ func (m *MySQLPlugin) FetchMetrics() (map[string]float64, error) {
...
 	}
-	db := mysql.New(proto, "", m.Target, m.Username, m.Password, "")
-	switch c := db.(type) {
-	case *native.Conn:
-		c.Debug = m.Debug
-	}
-	err := db.Connect()
+
+	// 	username:password@unix(/var/lib/mysql/mysql.sock)/dbname?param=value
+	dsn := fmt.Sprintf("%s:%s@%s(%s)/", m.Username, m.Password, proto, m.Target)
```
Comment (Arthur1) [inline lib/mysql.go:417]: There is a way to generate dns strings using `Config` from github.com/go-sql-driver/mysql package. It is not required, but may improve the outlook of the code.

```go
cnf := mysql.NewConfig()
cnf.User = m.Username
cnf.Passwd = m.Password
cnf.Net = proto
cnf.Addr = m.Target
// e.g.) "username:password@unix(var/lib/mysql/mysql.sock)"
dsn := cnf.FormatDSN()
```

**Review by Arthur1 — COMMENTED** [review]
Please see the point about the DEBUG option. The suggestion about how to make DSN is optional.

**Review by Arthur1 — APPROVED** [review]
lgtm

**Comment by yseto** [comment]
Thanks for your review.

**Pull request merged**